src/Symfony/Component/HttpKernel/ControllerMetadata/ControllerMetadataUtil.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\ControllerMetadata;
+
+use Doctrine\Common\Util\ClassUtils;
+
+abstract class ControllerMetadataUtil
+{
+    /**
+     * Gets a class name (if available) and method/function name from a callable
+     *
+     * @param mixed $controller
+     * @return string[] the class name and method name (where applicable)
+     */
+    public static function getControllerLogicalName($controller)
+    {
+        if ($controller instanceof \Closure) {
+            // cannot store any metadata of anonymous functions
+            return null;
+        }
+
+        if (is_object($controller)) {
+            // callable class
+            $controller = array($controller, '__invoke');
+        } elseif (is_string($controller)) {
+            // normal function
+            $controller = array(null, $controller);
+        }
+
+        $className = null;
+        $method = $controller[1];
```

Comment:
Actually thinking of changing the method typhint from mixed to callable, this way I don't have to check because it would not be considered a valid callable.